standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -287,29 +287,7 @@ PartitionExpressionProxy getExpressionProxy() {
     return expressionProxy;
   }
 
-  /**
-   * Use {@link #getThreadId()} instead.
-   * @return thread id
-   */
-  @Deprecated
-  public static Integer get() {
-    return HMSHandlerContext.getThreadId();
-  }
-
-  @Override
-  public int getThreadId() {
-    return HMSHandlerContext.getThreadId();
-  }
-
-  public HMSHandler(String name) throws MetaException {
-    this(name, MetastoreConf.newMetastoreConf(), true);
-  }
-
-  public HMSHandler(String name, Configuration conf) throws MetaException {
-    this(name, conf, true);
-  }
-
-  public HMSHandler(String name, Configuration conf, boolean init) throws MetaException {
```

Comment:
I am not sure about this change. Why did we remove these methods?

Reply:
In our production(standalone or embeded metastore),  the `init` is always set to false:
standalone: main -> https://github.com/apache/hive/blob/4909ba369cc331a1b490e23b73cbb2cb43f29bd9/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java#L314 -> https://github.com/apache/hive/blob/4909ba369cc331a1b490e23b73cbb2cb43f29bd9/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java#L370

embeded: https://github.com/apache/hive/blob/4909ba369cc331a1b490e23b73cbb2cb43f29bd9/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java#L118-L134

then we init the handler in RetryingHMSHandler,
https://github.com/apache/hive/blob/4909ba369cc331a1b490e23b73cbb2cb43f29bd9/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/RetryingHMSHandler.java#L80

Reply:
The `init` is true only in our tests,  so I propose to remove the parameter `init` here, so that we follow the same style, i.e, the handler can not be used until after the method `init` has been called.

Reply:
Hi @pvary, cloud that make sense, what do you think about it?

Reply:
Sorry @dengzhhu653, I was distracted by other tasks.
This makes sense